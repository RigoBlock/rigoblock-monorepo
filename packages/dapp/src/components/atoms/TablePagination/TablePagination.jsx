import './TablePagination.scss'
import Icon from '../Icon'
import PropTypes from 'prop-types'
import React from 'react'

const TablePagination = ({
  page,
  pages,
  onPageChange,
  canPrevious,
  canNext
}) => {
  const changePage = newPage =>
    page !== newPage ? onPageChange(newPage) : null
  return (
    <div className="pagination">
      <div className="table-navigation">
        <span className="first-page" onClick={() => changePage(0)}>
          <Icon type="navigate_before" className="stacked-icon" />
          <Icon type="navigate_before" />
        </span>
        <Icon
          type="navigate_before"
          onClick={() => (canPrevious ? changePage(page - 1) : null)}
        />
        <span className="page-number">{page + 1}</span>
        <Icon
          type="navigate_next"
          onClick={() => (canNext ? changePage(page + 1) : null)}
        />
        <span className="last-page" onClick={() => changePage(pages)}>
          <Icon type="navigate_next" />
          <Icon type="navigate_next" className="stacked-icon" />
        </span>
      </div>
    </div>
  )
}
TablePagination.propTypes = {
  page: PropTypes.number.isRequired,
  pages: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired,
  canNext: PropTypes.bool.isRequired,
  canPrevious: PropTypes.bool.isRequired
}

export default TablePagination
