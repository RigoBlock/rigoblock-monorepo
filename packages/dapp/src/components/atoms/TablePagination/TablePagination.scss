@import "../../_settings/variables.scss";

.pagination {
  background: $secondary-bg-color;
  display: flex;
  justify-content: flex-end;
  padding-top: $dist-s;
  width: 100%;

  .icon {
    color: $azure-radiance;
    user-select: none;
  }
  .page-number {
    line-height: $line-height;
    padding: 0 $dist-xs;
    user-select: none;
  }
  .table-navigation {
    display: flex;
    align-content: center;
  }

  .first-page {
    .stacked-icon {
      left: $dist-s;
      position: relative;
    }
  }

  .last-page {
    .stacked-icon {
      position: relative;
      right: $dist-s;
    }
  }
}
