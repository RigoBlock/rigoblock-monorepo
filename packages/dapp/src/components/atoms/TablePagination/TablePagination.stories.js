import '../../_settings/_base.scss'
import { MemoryRouter } from 'react-router-dom'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import React from 'react'
import TablePagination from './TablePagination'

const props = {
  page: 0,
  pages: 10,
  onPageChange: () => {},
  canPrevious: true,
  canNext: true
}

storiesOf('Atoms/TablePagination', module)
  .addDecorator(withKnobs)
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .add('default', () => <TablePagination {...props} />)
