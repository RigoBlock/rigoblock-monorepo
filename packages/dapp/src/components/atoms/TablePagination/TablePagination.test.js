import { shallow } from 'enzyme'
import TablePagination from './TablePagination'
import toJson from 'enzyme-to-json'

const props = {
  page: 0,
  pages: 10,
  onPageChange: () => {},
  canPrevious: true,
  canNext: true
}

describe('TablePagination component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(TablePagination, props)))
    ).toMatchSnapshot()
  })
})
