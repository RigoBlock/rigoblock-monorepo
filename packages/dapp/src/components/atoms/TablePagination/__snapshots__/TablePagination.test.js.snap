// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TablePagination component renders correctly 1`] = `
<div
  className="pagination"
>
  <div
    className="table-navigation"
  >
    <span
      className="first-page"
      onClick={[Function]}
    >
      <Icon
        className="stacked-icon"
        onClick={[Function]}
        size="medium"
        tooltipText=""
        type="navigate_before"
      />
      <Icon
        className=""
        onClick={[Function]}
        size="medium"
        tooltipText=""
        type="navigate_before"
      />
    </span>
    <Icon
      className=""
      onClick={[Function]}
      size="medium"
      tooltipText=""
      type="navigate_before"
    />
    <span
      className="page-number"
    >
      1
    </span>
    <Icon
      className=""
      onClick={[Function]}
      size="medium"
      tooltipText=""
      type="navigate_next"
    />
    <span
      className="last-page"
      onClick={[Function]}
    >
      <Icon
        className=""
        onClick={[Function]}
        size="medium"
        tooltipText=""
        type="navigate_next"
      />
      <Icon
        className="stacked-icon"
        onClick={[Function]}
        size="medium"
        tooltipText=""
        type="navigate_next"
      />
    </span>
  </div>
</div>
`;
